src/dispatch/incident/views.py

```diff
@@ -138,6 +138,19 @@ def update_incident(
     if not incident:
         raise HTTPException(status_code=404, detail="The requested incident does not exist.")
 
+    # we want to provide additional protections around restricted incidents
+    if incident.visibility == Visibility.restricted:
+        # reject if the user isn't an admin or commander
+        if incident.commander.email == current_user.email:
+            return incident
+
+        if current_user.role == UserRoles.admin:
+            return incident
```

Comment:
I think we want to run the update code here.

Reply:
Yup, I just noticed the error. Added a commit that fixes it.